### PR TITLE
Cleanup handling of podman mount/unmount

### PR DESCRIPTION
**Pull request by rhatdan**
We should default to the user name unmount rather then the internal
name of umount.

Also User namespace was not being handled correctly. We want to inform
the user that if they do a mount when in rootless mode that they have
to be first in the podman unshare state.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Splitting this PR away from the bigger podmain image mount PR.  Hopefully this will help me figure out what is causing the podman tag/untag issues, that are preventing that PR from merging.

File: pkg/rootless/rootless_linux.c [hunk 1]
```diff
@@ -210,6 +210,13 @@ can_use_shortcut ()
           ret = false;
           break;
         }
+
+      if (strcmp (argv[argc], "container") == 0 &&
+	   strcmp (argv[argc+1], "mount") == 0)
```
Comment (edsantiago) [inline pkg/rootless/rootless_linux.c:215]: index out of range

Comment (rhatdan) [reply pkg/rootless/rootless_linux.c:215]: Fixed

File: docs/source/markdown/links/podman-unmount.1 [hunk 2]
```diff
@@ -1 +0,0 @@
-.so man1/podman-umount.1
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/links/podman-unmount.1:1]: Was this intentionally deleted?

**Comment by TomSweeneyRedHat** [comment]
One question, and tests as @vrothberg mentioned would be good.  Otherwise LGTM

File: docs/source/markdown/links/podman-unmount.1 (hunk 2 above)
Comment (rhatdan) [reply docs/source/markdown/links/podman-unmount.1:1]: Yes, It is actually renamed.  podman-umount.1 now links to man1/podman-unmount.1

**Comment by rhatdan** [comment]
Tests added, but I could not get them to run locally.  Will see what happens here.

File: cmd/podman/registry/config.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ import (
 
 const (
 	ParentNSRequired = "ParentNSRequired"
+	UserNSRequired   = "UserNSRequired"
```
Comment (mheon) [inline cmd/podman/registry/config.go:19]: Could this be renamed? Maybe "RootRequired"?

Comment (rhatdan) [reply cmd/podman/registry/config.go:19]: I don't think that is helpful.  How about UnshareUserNSRequired Or UnshareNSRequired

Comment (mheon) [reply cmd/podman/registry/config.go:19]: UnshareNSRequired SGTM

**Comment by mheon** [comment]
LGTM once the UserNSRequired field is renamed

**Comment by vrothberg** [comment]
Restarted flakes

**Pull request merged**